### DISCUSSION TITLE
Custom Bit Width

### DISCUSSION THREAD

**Discussion by leonuetzel** [flandreas/antares#286] [Q&A] [answered]
Hello,

im wondering how i can set the Bus Width of Signals inside a Circuit to a Number not predefined.
I guess i gotta use the "Expression"-Option but it requires a Script to enter the Number and i dont know how to script that.

**Comment by flandreas** [comment, marked as answer]
Unfortunately, that's not yet possible. You can currently only have one of the predefined bit widths. Even with scripting: Every script that determines a bit width must yield one of the predefined bit widths.

The reason for this restriction was originally the challenge of displaying hexadecimal numbers in nibbles (chunks of 4 bits), so (with the exception of 2) only multiples of 4 were supported as bit width. However, this restriction has recently been relaxed with the introduction of 3 and 5.

Which bit widths are you missing?

**Reply by leonuetzel** [reply]
Ah ok, i was just trying something with 40 and 48 Bits, but i can implement it with 32+8 and 32+16.
